aiohttp/web_response.py

```diff
@@ -268,6 +271,41 @@ def last_modified(
         elif isinstance(value, str):
             self._headers[hdrs.LAST_MODIFIED] = value
 
+    @property
+    def etag(self) -> Optional[ETag]:
+        quoted_value = self._headers.get(hdrs.ETAG)
+        if not quoted_value:
+            return None
+        elif quoted_value == ETAG_ANY:
+            return ETag(value=ETAG_ANY)
+        match = QUOTED_ETAG_RE.match(quoted_value)
+        if not match:
+            return None
+        is_weak, value = match.group(1), match.group(2)
+        return ETag(
+            is_weak=bool(is_weak),
+            value=value[1:-1],
+        )
+
+    @etag.setter
+    def etag(self, value: Optional[Union[ETag, str]]) -> None:
+        if (isinstance(value, str) and value == ETAG_ANY) or (
+            isinstance(value, ETag) and value.value == ETAG_ANY
+        ):
+            self._headers[hdrs.ETAG] = ETAG_ANY
+        elif isinstance(value, str):
+            self._headers[hdrs.ETAG] = f'"{value}"'
+        elif isinstance(value, ETag) and isinstance(value.value, str):
+            hdr_value = f'W/"{value.value}"' if value.is_weak else f'"{value}"'
+            self._headers[hdrs.ETAG] = hdr_value
+        elif value is None:
+            self._headers.pop(hdrs.ETAG, None)
+        else:
+            raise ValueError(
+                f"Unsupported etag type: {type(value)!r}. "
```

Comment:
`{type(value)!r}` is the same as `type(value)`. Maybe you meant `{type(value).__name__!r}`?

Reply:
Yeah, that's exactly what I meant. I guess better solution will be remove `!r`, because library code mostly contains `'%r' % type(smth)`.